src/Sylius/Bundle/AdminBundle/Twig/Component/Shared/Navbar/NotificationsComponent.php

```diff
@@ -13,16 +13,89 @@
 
 namespace Sylius\Bundle\AdminBundle\Twig\Component\Shared\Navbar;
 
+use Psr\Clock\ClockInterface;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Sylius\Bundle\CoreBundle\SyliusCoreBundle;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
 use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
 
 class NotificationsComponent
 {
-    /**
-     * @return array<string, mixed>
-     */
+    public const LATEST_SYLIUS_VERSION_CACHE_KEY = 'latest_sylius_version';
+
+    public function __construct(
+        private readonly ClientInterface $client,
+        private readonly RequestStack $requestStack,
+        private readonly RequestFactoryInterface $requestFactory,
+        private readonly StreamFactoryInterface $streamFactory,
+        private readonly CacheInterface $cache,
+        private readonly ClockInterface $clock,
+        private readonly string $hubUri,
+        private readonly string $environment,
+    ) {
+    }
+
+    /** @return array<array-key, mixed> */
     #[ExposeInTemplate(name: 'notifications')]
     public function getNotifications(): array
     {
-        return [];
+        $metadata = $this->cache instanceof ItemInterface ? $this->cache->getMetadata() : [];
+        $metadata[ItemInterface::METADATA_EXPIRY] = $this->clock->now()->modify('+1 day')->getTimestamp();
```

Comment:
```suggestion
        $metadata[ItemInterface::METADATA_EXPIRY] = $this->clock->now()->modify(sprintf('+%d minutes', $frequency))->getTimestamp();
```
we have a `frequency` setting regarding this notification in the Admin's configuration (I think it's in minutes 🤔?)